internal/pkg/aws/route53/domain.go

```diff
@@ -30,22 +30,26 @@ func NewRoute53Domains(s *session.Session) *Route53Domains {
 	}
 }
 
-// IsDomainOwned checks if the domain is owned by the account.
-func (r *Route53Domains) IsDomainOwned(domainName string) error {
+// IsDomainRegisteredInRoute53 checks if the domain is owned by the account.
+func (r *Route53Domains) IsDomainRegisteredInRoute53(domainName string) error {
```

Comment:
nit: since this is the route53 client we probably don't need to repeat route53 in the method name:
```suggestion
func (r *Route53Domains) IsRegisteredDomain(domainName string) error {
```